verifyFiles.py

```diff
@@ -157,10 +185,17 @@ def n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_fro
 		    elif conditions_dict[c][subject][file_name].st_size != conditions_dict[d][subject][file_name].st_size :
 		      diff[key][file_name]+=1
                       files_are_different=True
-	   	    elif checksum(abs_path_c) != checksum(abs_path_d):# TODO:when there are multiple conditions, we will compute checksums multiple times.We should avoid that.
+	   	    elif abs_path_c in dictionary_checksum and  abs_path_d in dictionary_checksum:# Added dictionary to avoid computing checksums multiple times.
 		      #Condition below makes sure that the checksums in the file after processing and in local are equal
-		      diff[key][file_name]+=1
-                      files_are_different=True
+		      if dictionary_checksum[abs_path_c] != dictionary_checksum[abs_path_d]:
+		        diff[key][file_name]+=1
+                        files_are_different=True
+		    else:
+			dictionary_checksum[abs_path_c] = checksum(abs_path_c)
+			dictionary_checksum[abs_path_d] = checksum(abs_path_d)
+			if dictionary_checksum[abs_path_c] != dictionary_checksum[abs_path_d]:
+                          diff[key][file_name]+=1
+                          files_are_different=True
```

Comment:
The code in lines 196-198 should be factorized with lines 190-192 (identical code).